### ISSUE TITLE
[ASCollectionNode] Change collectionView:constrainedSizeForNodeAtIndexPath: to call on main.

### ISSUE THREAD

**Issue by Adlai-Holler** [facebook/AsyncDisplayKit#1260]
The current implementation of this method inside `ASPagerNode.m` is:

```objectivec
- (ASSizeRange)collectionView:(ASCollectionView *)collectionView constrainedSizeForNodeAtIndexPath:(NSIndexPath *)indexPath
{
  if (_pagerDataSourceImplementsConstrainedSizeForNode) {
    return [_pagerDataSource pagerNode:self constrainedSizeForNodeAtIndexPath:indexPath];
  }
  return ASSizeRangeMake(CGSizeZero, self.view.bounds.size);
}
```

ASDataController calls this method off the main thread, but this method may access `self.view.bounds.size` which is a violation of UIKit's main-thread requirement. @levi Thoughts?

**Comment by Adlai-Holler** [comment]
Also, should we return min.width=self.width instead of 0? If any of ASPagerNode's cells is narrower than the node itself, the paging won't work as expected.

**Comment by appleguy** [comment]
@Adlai-Holler thanks for reporting this. Ironically, before I read this issue, I just recently discovered that we were calling this method off of the main thread. I think this is a much more important issue than just the pager implementation, because we are passing an index Path here and so there is a data model consistency issue just as there would be if we requested actual nodes in this way.

I think it is rare in practice for developers to have an implementation of this method where this is a critical problem, but for those who do, it could result in a pretty serious issue.  @nguyenhuy is currently working on a fix, but we will see if it gets into the next release. I have a pull request up that Restore is the behavior of previous versions of the framework, but it does not fix this particular issue.

I agree with you that we can change the minimum value to be the same as the maximum so that the page size is by default guaranteed to match the bound size.
